File: tools/push_firebase.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

void main() async {
  final version = '26.08.09.d.beta';
  final notes = 'Sửa lỗi tự nhảy tập khi Stremio đang tải bộ đệm';
  final channel = 'beta';
  
  final url = Uri.parse('https://firestore.googleapis.com/v1/projects/tv4u-ec4ae/databases/(default)/documents/updates/$channel');
  
  final body = {
    'fields': {
      'latest_version': {'stringValue': version},
      'release_notes': {'stringValue': notes},
      'download_url': {'stringValue': 'https://github.com/Tien360/mytv4u/releases/download/$version/MyTV4U_Setup_$version.exe'},
      'is_force_update': {'booleanValue': false},
      'release_date': {'timestampValue': DateTime.now().toUtc().toIso8601String()},
    }
  };

  print('Đang cập nhật Firebase ($channel)...');
  final res = await http.patch(
    url, 
    headers: {'Content-Type': 'application/json; charset=UTF-8'},
    body: json.encode(body)
  );
  
  if (res.statusCode == 200) {
    print('Thành công!');
  } else {
    print('Lỗi Firebase: ${res.statusCode} ${res.body}');
  }
}

File: push_firebase.py
import json
import urllib.request
import urllib.parse
import datetime

version = "26.08.31.20.public"
notes = "- Tối ưu UI Cài đặt: đồng bộ giao diện Ambient Background mượt mà.\n- Sửa lỗi cấu trúc UI của thẻ cài đặt Nhạc và Dropdown.\n- Bổ sung tuỳ chọn Bật/Tắt Giao diện Tối giản trong Cài đặt.\n- Cập nhật Trợ lý Tối ưu hoá gợi ý cho máy yếu/dùng pin."

url = "https://firestore.googleapis.com/v1/projects/tv4u-ec4ae/databases/(default)/documents/updates/public"
data = {
    "fields": {
      "latest_version": {"stringValue": version},
      "release_notes": {"stringValue": notes},
      "download_url": {"stringValue": f"https://github.com/Tien360/mytv4u/releases/download/{version}/MyTV4U_Setup_{version}.exe"},
      "is_force_update": {"booleanValue": False},
      "release_date": {"timestampValue": datetime.datetime.utcnow().isoformat() + "Z"}
    }
}
req = urllib.request.Request(url, data=json.dumps(data).encode("utf-8"), method="PATCH")
req.add_header("Content-Type", "application/json; charset=UTF-8")

try:
    with urllib.request.urlopen(req) as response:
        print("Cập nhật Firebase thành công!")
except urllib.error.HTTPError as e:
    print(f"Lỗi: {e.code} {e.read().decode('utf-8')}")
